mqtt2kasa/log.py

```diff
@@ -17,42 +20,43 @@ def _log_handler_address(files=tuple()):
         )
         return None
 
+def is_running_in_docker():
+    return path.exists('/.dockerenv')
 
 def initLogger(testing=False):
     logger = getLogger()
-    logger.setLevel(logging.INFO)
-
-    format = (
-        "%(asctime)s [mqtt2kasa] %(module)12s:%(lineno)-d %(levelname)-8s %(message)s"
-    )
-    formatter = logging.Formatter(format)
+    logger.setLevel(logging.INFO)        
 
-    # Logs are normally configured here: /etc/rsyslog.d/*
-    logHandlerAddress = _log_handler_address(
-        ["/run/systemd/journal/syslog", "/var/run/syslog", "/device/log"]
-    )
+    if is_running_in_docker():
+        log_to_console()
+        getLogger().info("Running in a Docker container. Logs are sent to stdout only")
+    else:
+        format = (
+            "%(asctime)s [mqtt2kasa] %(module)12s:%(lineno)-d %(levelname)-8s %(message)s"
+        )
+        formatter = logging.Formatter(format)
 
-    if logHandlerAddress:
-        syslog = SysLogHandler(
-            address=logHandlerAddress, facility=SysLogHandler.LOG_DAEMON
+        # Logs are normally configured here: /etc/rsyslog.d/*
+        logHandlerAddress = _log_handler_address(
+            ["/run/systemd/journal/syslog", "/var/run/syslog", "/device/log"]
         )
-        syslog.setFormatter(formatter)
-        logger.addHandler(syslog)
-    else:
-        stdout_handler = logging.StreamHandler()
```

Comment:
in case of any issues with `["/run/systemd/journal/syslog", "/var/run/syslog", "/device/log"]`, the fallback strategy was to output to console but using the `"%(asctime)s [mqtt2kasa] %(module)12s:%(lineno)-d %(levelname)-8s %(message)s"` which resulted in multiline comments when I tested. I made it consistent with default console format `%(asctime)s %(module)12s:%(lineno)-d %(levelname)-8s %(message)s`